ee/clickhouse/queries/session_recordings/test/test_clickhouse_session_recording_list.py

```diff
@@ -76,3 +76,28 @@ def test_event_filter_with_cohort_properties(self):
                 (session_recordings, _) = session_recording_list_instance.run()
                 self.assertEqual(len(session_recordings), 1)
                 self.assertEqual(session_recordings[0]["session_id"], "2")
+
+    @freeze_time("2021-01-21T20:00:00.000Z")
+    @snapshot_clickhouse_queries
+    @test_with_materialized_columns(["$current_url", "$session_id"], verify_no_jsonextract=False)
+    def test_event_filter_with_matching_on_session_id(self):
+        Person.objects.create(team=self.team, distinct_ids=["user"], properties={"email": "bla"})
+        self.create_snapshot("user", "1", self.base_time, window_id="1")
+        self.create_event("user", self.base_time, properties={"$session_id": "1"})
+        self.create_event("user", self.base_time, event_name="$autocapture", properties={"$session_id": "2"})
+        self.create_snapshot("user", "1", self.base_time + relativedelta(seconds=30), window_id="1")
+        filter = SessionRecordingsFilter(
+            team=self.team, data={"events": [{"id": "$pageview", "type": "events", "order": 0, "name": "$pageview"}]},
+        )
```

Comment:
I'm a little confused here, how do we get the events belonging to a specific `session_id` if that `session_id` isn't specified in the Filter? Totally might be missing something here

Reply:
It's a bit confusing, but the query here is: "Show me all recordings that contain the `$pageview` event". The change is that we now say a recording "contains" an event if they share a `session_id`, and before it was based on a timestamp.

So the really interesting part of this test is second piece where the `$autocapture` event is within the timebounds of the recording, but because it has a different `session_id` that recording is not returned.

Does that make sense?

Reply:
Just added a comment to make this clearer.